mono/metadata/threads-types.h

```diff
@@ -189,7 +189,7 @@ void ves_icall_System_Runtime_Remoting_Contexts_Context_ReleaseContext (MonoAppC
 
 MonoInternalThread *mono_thread_internal_current (void);
 
-void mono_thread_internal_abort (MonoInternalThread *thread);
+void mono_thread_internal_abort (MonoInternalThread *thread, gboolean appdomain_unload);
```

Comment:
You could create a `mono_thread_internal_abort_appdomain_unload` function instead of adding a parameter which is going to be `FALSE` all of the time except for 1 case.